src/crafting_gui.cpp

```diff
@@ -226,7 +226,7 @@ const recipe *select_crafting_recipe( int &batch_size )
 
             TAB_MODE m = ( batch ) ? BATCH : ( filterstring.empty() ) ? NORMAL : FILTERED;
             draw_recipe_tabs( w_head, tab.cur(), m );
-            draw_recipe_subtabs( w_subhead, tab.cur(), subtab.cur(), m );
+            draw_recipe_subtabs( w_subhead, tab.cur(), subtab.cur(), available_recipes );
```

Comment:
Why did you remove the argument `m` from the function call? As this is the only place that calls the function, it's never called with an explicit value for that `TAB_MODE mode` parameter (the default value will be used instead). You might as well remove that parameter from the function as it's now useless.